src/lib.rs

```diff
@@ -198,12 +201,13 @@ pub use poll::Iter as EventsIter;
 #[doc(hidden)]
 pub use io::deprecated::would_block;
 
-#[cfg(all(unix, not(target_os = "fuchsia")))]
+#[cfg(any(all(unix, not(target_os = "fuchsia")), target_os = "redox"))]
```

Comment:
Is it possible to move this to a separate block? So, split out the `cfg(redox)` and have a separate `pub mod unix`? It would make it easier to manage IMO.